builder/azure/common/template/template_builder.go

```diff
@@ -636,6 +643,43 @@ const KeyVault = `{
   ]
 }`
 
+// adapted from https://github.com/Azure/azure-quickstart-templates/blob/b78db5d27fd6344656ebdd30f74eee5d22f40dde/application-workloads/ros/ros-vm-windows/nestedtemplates/customScriptExtension.json
+const CustomScript = `{
+    "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "vmName": {
+            "metadata": {
+                "description": "The name of the vm"
+            },
+            "type": "string"
+        },
+        "commandToExecute": {
+            "type": "string"
+        }
+    },
+    "variables": {
+        "location": "[resourceGroup().location]"
+    },
+    "resources": [
+        {
+            "type": "Microsoft.Compute/virtualMachines/extensions",
+            "apiVersion": "2019-12-01",
```

Comment:
the api-version seems quite old. the latest one is here: https://learn.microsoft.com/en-us/rest/api/compute/virtual-machine-extensions/create-or-update?tabs=HTTP

Reply:
Done. I've updated it to `2022-08-01`.
Parameters (that I'm using at least) seem to be the same.